corehq/apps/app_manager/suite_xml/post_process/instances.py

```diff
@@ -160,9 +165,20 @@ def require_instances(entry, instances=(), instance_ids=()):
             entry.instances = sorted_instances
 
 
-def get_instance_factory(scheme):
-    return get_instance_factory._factory_map.get(scheme, preset_instances)
-get_instance_factory._factory_map = {}
+_factory_map = {}
+
+
+def get_instance_factory(instance_name):
+    try:
+        scheme, _ = instance_name.split(':', 1)
+    except ValueError:
+        scheme = instance_name
+
+    return _factory_map.get(scheme, null_factory)
```

Comment:
Prior to this, if the scheme did not contain a ':' it would be set to `None` (unless `instance_name = 'location').
Then, when getting the instance factory, if no factory was found for the scheme (which would be the case if scheme = None) then the 'preset_instances' factory was used:
```python
try:
        scheme, _ = instance_name.split(':', 1)
    except ValueError:
        scheme = instance_name if instance_name == 'locations' else None

    return _factory_map.get(scheme, preset_instances)
```

If the preset_instances factory could not handle the `instance_name` it would return None. 

This change makes that clearer by assuming that 'instance_name' IS the scheme if it does not contain a ':'. If no instance factory is registered to handle the scheme it will return the `null_factory`.